src/main/java/com/softserve/service/impl/UserServiceImpl.java

```diff
@@ -186,6 +187,7 @@ public User registration(User user) {
      * @return User entity
      * @throws IncorrectPasswordException when password is incorrect or not strong enough
      */
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
```

Comment:
This will solve only one case failure. Possible data inconsistancy

Reply:
In other cases of propagation we got or nullException with user{id=null} or ConstraintViolationException and rollback transactions with GlobalExceptionHandler - java.io.IOException. So each way we want to autoregister a user, we require a new transaction without interrupting other.